R/batch.R

```diff
@@ -148,20 +138,31 @@ generate_batch_maker <- function(run_length, where_the_inputs_at, args, size) {
       args[[j]] <- second_arg
     }
     i <<- i + size
-    list('new_call' = args, 'keys' = keys, 'num_batches' = ceiling(run_length / size))
+    list("new_call" = args, "keys" = keys, "num_batches" = ceiling(run_length / size))
   }
 }
 
-default_batch_error <- function(e, stop, batchman.verbose) {
+iterated_try_catch <- function(expr, stop, retry, new_call, run_env) {
+  tryCatch(
+    eval(new_call, envir = run_env),
+    error = function(e) {
+      raise_error_or_warning(e, stop, batchman.verbose)
+      if (retry > 0) { iterated_try_catch(expr, stop, retry - 1, new_call, run_env) }
+      else { NA }
```

Comment:
`else { base::stop("Batching failed") }` to comply with `It should try retry times, but if all those fail then the whole batching should fail.`
I agree that this is the correct behavior. You can hide the existing behavior behind another feature parameter, like `forgiving = TRUE`, but I agree that the default behavior should be failing if at least one batch fails